dcc-portal-ui/app/scripts/downloader/js/request.js

```
@@ -136,8 +136,10 @@
 
         $scope.params.processing = false;
       },function (error) {
-        $scope.params.processing = false;
-        $scope.params.downloadEnabled = false;
+        if(error.status === 503){
+          $scope.params.processing = false;
+          $scope.params.downloadEnabled = false;
+        }
```

Perhaps we should propagate the error if not 503?

Also, I would move `$scope.params.processing = false;` to a `finally` handler.

---

@cheapsteak What's the best way to trigger the global error handler for restangular if not a `503` above? `throw error`?

---

I'm not sure if I'm understanding correctly, but I think our app's `setErrorInterceptor` ([here](https://github.com/icgc-dcc/dcc-portal/blob/41fe22825c3c1edef43d6dc53315ac558f606bc1/dcc-portal-ui/app/scripts/app/js/app.js#L602-L614)) would receive the error before this error handler does

I think we may want to change its current check of `if (response.status === 500) {` to `if (_.inRange(response.status, 500, 600)) {`

---

The semantics that we want are the ability to override the global error interceptor locally.

FYI - docs on semantics here: https://github.com/mgonto/restangular#seterrorinterceptor (reading now)

---

Not sure I understand this:

> The errorInterceptor function, whenever it returns false, prevents the promise linked to a Restangular request to be executed. All other return values (besides false) are ignored and the promise follows the usual path, eventually reaching the success or error hooks.

Also, not really sure what `responseHandler` is. Is that the local error handler?

---

Discussion here:

http://stackoverflow.com/questions/28010548/restangular-how-to-override-error-interceptors

---

Also, we are not explicitly returning `true` or `false` from:

https://github.com/icgc-dcc/dcc-portal/blob/41fe22825c3c1edef43d6dc53315ac558f606bc1/dcc-portal-ui/app/scripts/app/js/app.js#L602-L614

Does that mean `false`? If it does, it stops the promise chain and `finally`  handlers etc. won't be called.

---

I think this maybe the only way:

https://github.com/mgonto/restangular/issues/566#issuecomment-34312971

see 

https://github.com/mgonto/restangular/issues/566#issuecomment-34453140